File: Easy/387-first-unique-character-in-a-string/try-6.cpp
/*
 * Problem #387: First Unique Character in a String
 * Difficulty: Easy
 * Submission: Try 6
 * status: Accepted
 * Language: cpp
 * Date: 06/12/2024, 22:52:15
 * Link: https://leetcode.com/problems/first-unique-character-in-a-string/
 */

class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> vec(26,0);
        for(auto i : s){
            vec[i-'a']++;
        }
        for(int i=0;i<s.size();i++){
            if(vec[s[i]-'a']==1){
                return i;
            }
        }
        return -1;
    }
};

File: Medium/424-longest-repeating-character-replacement/try-6.py
/*
 * Problem #424: Longest Repeating Character Replacement
 * Difficulty: Medium
 * Submission: Try 6
 * status: Accepted
 * Language: python3
 * Date: 08/12/2024, 16:58:55
 * Link: https://leetcode.com/problems/longest-repeating-character-replacement/
 */

class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        d={}
        mx=0
        start=0
        re=0
        for end in range(len(s)):
            if s[end] in d:
                d[s[end]]+=1
            else:
                d[s[end]]=1
            mx=max(mx,d[s[end]])
            while(end-start+1-mx>k):
                d[s[start]]-=1
                start+=1
            
            re=max(re,end-start+1)
        return re
        
